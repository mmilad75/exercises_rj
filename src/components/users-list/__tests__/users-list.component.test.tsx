import { render, screen, waitFor } from "@testing-library/react";
import mockAxios from "jest-mock-axios";
import React from "react";
import UsersList from "../users-list.component";
import users from "../mockUsersList";

describe("UsersList component", () => {
    afterEach(() => {
        mockAxios.reset();
    });

    it("should render correctly", () => {
        jest.spyOn(React, "useEffect").mockImplementationOnce(() => {});
        const { asFragment } = render(<UsersList />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should display loading message when there is no usersFetched", () => {
        jest.spyOn(React, "useEffect").mockImplementationOnce(() => {});
        render(<UsersList />);

        const messageElement = screen.getByText("Loading users...");

        expect(messageElement).toBeInTheDocument();
    });

    it("should display correct number of users", async () => {
        mockAxios.get.mockResolvedValue({ data: users });
        jest.spyOn(React, "useEffect").mockImplementationOnce((callback) => {
            callback();
        });

        render(<UsersList />);

        await waitFor(() => {
            expect(screen.getAllByRole("listitem").length).toBe(users.length);
        });
    });
});
