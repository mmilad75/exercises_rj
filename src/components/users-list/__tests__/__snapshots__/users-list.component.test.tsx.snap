// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Users List
    </h1>
    <p>
      Loading users...
    </p>
  </div>
</DocumentFragment>
`;
